debugtools/DDR_VM/src/com/ibm/j9ddr/vm29/tools/ddrinteractive/gccheck/CheckEngine.java

```diff
@@ -1189,6 +1194,29 @@ private int checkJ9Object(J9ObjectPointer object, GCHeapRegionDescriptor regionD
 				return ret;
 			}
 		}
+
+		if (GCExtensions.isVLHGC() && J9BuildFlags.env_data64 && isIndexableDataAddressFlagSet() && ObjectModel.isIndexable(object)) {
```

Comment:
Is there a reason why check is limited to Balanced only? I know it supposed to be used in Balanced for now but is something prevents is to do this check for Standard collectors? We should update dataAddr properly too. Metronome never moves objects, so initial value should be correct also. I would prefer not to check for GC policy here if it is possible.

Reply:
I don't believe there should be anything preventing us from doing this check in standard collectors. I will test this to make sure.